core/test/unit/middleware/cors_spec.js

```diff
@@ -137,4 +141,41 @@ describe('cors', function () {
 
         done();
     });
+
+    it('should be enabled if the origin matches config.url', function (done) {
+        var origin = 'http://my.blog';
+        configUtils.set({
+            url: origin
+        });
+
+        req.get = sinon.stub().withArgs('origin').returns(origin);
+        res.get = sinon.stub().withArgs('origin').returns(origin);
+        req.headers.origin = origin;
+
+        cors(req, res, next);
+
+        next.called.should.be.true();
+        res.headers['Access-Control-Allow-Origin'].should.equal(origin);
+
+        done();
+    });
```

Comment:
> We really shouldn't be sending `Access-Control-Allow-Origin` on same-origin requests.

I'm not sure if that's what the test reflects? It seems to me that nothing in the test or the middleware (not even cors itself, which is surprising) does any checks to determine if the request is cross-origin. In the tests we don't define the request destination at all.

As far as I understood, whenever I made a same-origin request, origin wasn't present in the headers and so cors was disabled.